src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+> where
+    <<E1<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+    <<E2<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+{
+    Nova {
+        args: LurkData<F>,
+        proof: nova::Proof<'a, F, C, M>,
+        num_steps: usize,
+    },
+}
+
+impl<
+        'a,
+        F: CurveCycleEquipped,
+        C: Coprocessor<F> + 'a + Serialize + DeserializeOwned,
+        M: MultiFrameTrait<'a, F, C>,
+    > HasFieldModulus for ProtocolProof<'a, F, C, M>
+where
+    <<E1<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+    <<E2<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+{
+    fn field_modulus() -> String {
+        F::MODULUS.to_owned()
+    }
+}
+
+impl MetaCmd<F> {
+    /// Returns the protocol variables, body and reduction count
+    ///
+    /// # Errors
+    /// * If the protocol evaluation fails
+    /// * If the variables aren't encoded in a proper list
+    /// * If the reduction count is not a number or can't be converted to `u64`
+    fn get_vars_body_and_rc(repl: &Repl<F>, ptcl: Ptr) -> Result<(Vec<Ptr>, Ptr, usize)> {
+        let (io, ..) = repl
+            .eval_expr(ptcl)
+            .with_context(|| "evaluating protocol")?;
+        let ptcl = &io[0];
+
+        let (vars, rest) = repl.store.car_cdr(ptcl)?;
+
+        let Some((vars, None)) = repl.store.fetch_list(&vars) else {
+            bail!("Protocol variables must be a proper list")
+        };
+
+        let (body, rest) = repl.store.car_cdr(&rest)?;
+
+        let (Ptr::Atom(Tag::Expr(ExprTag::Num), idx), _) = repl.store.car_cdr(&rest)? else {
+            bail!("Reduction count must be a Num")
+        };
+        let Some(rc) = repl.store.expect_f(idx).to_u64().map(|u| u as usize) else {
+            bail!("Invalid value for reduction count")
+        };
+        Ok((vars, body, rc))
+    }
+
+    /// Returns a vector containing the elements of a cons-list.
+    ///
+    /// # Errors
+    /// Errors if the the list is not proper or if the number of elements doesn't
+    /// match the expected `len`
+    fn get_args_vec(repl: &Repl<F>, args: &Ptr, len: usize) -> Result<Vec<Ptr>> {
+        let Some((args_vec, None)) = repl.store.fetch_list(args) else {
+            bail!("Protocol arguments must be a proper list")
+        };
+
+        if args_vec.len() != len {
+            bail!("Incompatible number of arguments. Protocol expects {len}")
+        }
+        Ok(args_vec)
+    }
+
+    /// Runs the protocol `body` with an environment built from the bindings
+    /// creted with `vars` and `args` and returns the proof input and the
+    /// post-verify function.
+    ///
+    /// # Errors
+    /// * If the protocol body evaluation fails
+    /// * If the protocol body doesn't evaluate to a pair
+    /// * If the proof input can't be built (first component of the pair is nil)
+    /// * If the proof input is not a proper list with length 6
+    fn get_cek_io_and_post_verify_function(
```

Comment:
This seems to be simulating the protocol function, when instead that function should actually have been created and be called here.

Reply:
What do you mean? It's not simulating. It's calling it for real.

But since we have the explicit arguments and body, there's no need to have an actual function. We just call the body with the correct environment.

Reply:
Yes, you just described simulating it.